shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -173,34 +182,14 @@ Engine::Engine(Delegate& delegate,
   environment->GetServices(parent_environment_service_provider.NewRequest());
   environment.Unbind();
 
-  OnEnableWireframe on_enable_wireframe_callback = std::bind(
-      &Engine::DebugWireframeSettingsChanged, this, std::placeholders::_1);
-
-  OnCreateView on_create_view_callback =
-      std::bind(&Engine::CreateView, this, std::placeholders::_1,
-                std::placeholders::_2, std::placeholders::_3);
-
-  OnUpdateView on_update_view_callback =
-      std::bind(&Engine::UpdateView, this, std::placeholders::_1,
-                std::placeholders::_2, std::placeholders::_3);
-
-  OnDestroyView on_destroy_view_callback =
-      std::bind(&Engine::DestroyView, this, std::placeholders::_1);
-
-  OnCreateSurface on_create_surface_callback =
-      std::bind(&Engine::CreateSurface, this);
-
-  // SessionListener has a OnScenicError method; invoke this callback on the
-  // platform thread when that happens. The Session itself should also be
-  // disconnected when this happens, and it will also attempt to terminate.
-  fit::closure on_session_listener_error_callback =
-      [dispatcher = async_get_default_dispatcher(),
-       weak = weak_factory_.GetWeakPtr()]() {
-        async::PostTask(dispatcher, [weak]() {
-          if (weak) {
-            weak->Terminate();
-          }
-        });
+  // SessionListener (and thus PlatformView) has a OnScenicError method; invoke
+  // this callback when that happens.  The OnScenicError callback occurs on the
+  // platform thread so there is no need to re-thread it.
```

Comment:
Just curious if there's a thread_id facility to DCHECK in the error closure.

Reply:
Yes I think so, fml::TaskRunner provides such things